src/client/Microsoft.Identity.Client/ManagedIdentity/V2/HybridCache.cs

```diff
@@ -0,0 +1,663 @@
+                var removedFromMemory = s_memoryCache.TryRemove(keyString, out _);
+                if (removedFromMemory)
+                {
+                    _logger.Verbose(() => $"[HybridCache] Token removed from memory cache for key: {key}");
+                }
+
+                // Step 2: Remove from file cache for persistence
+                await ExecuteWithMutexAsync(async () =>
+                {
+                    var fileCache = await LoadFileCacheAsync().ConfigureAwait(false);
+
+                    // Only save if we actually removed something
+                    if (fileCache.Remove(keyString))
+                    {
+                        await SaveFileCacheAsync(fileCache).ConfigureAwait(false);
+                        _logger.Verbose(() => $"[HybridCache] Token removed from file cache for key: {key}");
+                    }
+
+                    return Task.CompletedTask;
+                }, ct).ConfigureAwait(false);
+
+                _logger.Info(() => $"[HybridCache] Remove operation completed for key: {key}");
+            }
+            finally
+            {
+                semaphore.Release();
+            }
+        }
+
+        /// <summary>
+        /// Executes an action while holding the named mutex for cross-process synchronization.
+        /// </summary>
+        /// <typeparam name="T">The return type of the action.</typeparam>
+        /// <param name="action">The asynchronous action to execute under mutex protection.</param>
+        /// <param name="ct">A cancellation token that can be used to cancel the operation.</param>
+        /// <returns>A task containing the result of the action.</returns>
+        /// <remarks>
+        /// This method:
+        /// - Attempts to acquire the named mutex with a 30-second timeout
+        /// - Executes the provided action while holding the mutex
+        /// - Ensures the mutex is always released, even if the action throws an exception
+        /// - Provides cancellation support through the cancellation token
+        /// 
+        /// The mutex timeout prevents deadlocks in case of abandoned mutexes or long-running operations.
+        /// </remarks>
+        /// <exception cref="TimeoutException">
+        /// Thrown when the mutex cannot be acquired within the 30-second timeout period.
+        /// </exception>
+        /// <exception cref="OperationCanceledException">
+        /// Thrown when the operation is canceled via the cancellation token.
+        /// </exception>
+        private async Task<T> ExecuteWithMutexAsync<T>(Func<Task<T>> action, CancellationToken ct)
+        {
+            bool mutexAcquired = false;
+            try
+            {
+                _logger.Verbose(() => $"[HybridCache] Attempting to acquire mutex: {_mutexName}");
+
+                // Try to acquire mutex with timeout to avoid deadlocks
+                mutexAcquired = _namedMutex.WaitOne(_mutexTimeout);
```

Comment:
This is where all the unit tests in HybridCacheTests fail. It seems the mutext can't be acquired multiple times in the same test.

See the test: `GetAsync_ValidCachedToken_ReturnsToken`

Reply:
CC @gladjohn, can you experiment with this when you get time?